applications/FluidDynamicsApplication/custom_elements/two_fluid_navier_stokes.cpp

```diff
@@ -1803,7 +1854,7 @@ void TwoFluidNavierStokes<TElementData>::CondenseEnrichment(
     MatrixType &rKeeTot,
     const VectorType &rRHSeeTot)
 {
-    const double min_area_ratio = -1e-6;
+    const double min_area_ratio = 1e-7;
```

Comment:
The value resulted from a series of numerical experiments. The set-up was chosen such that the water surface coincided with element boundaries. The following parameters were considered:

- spurious velocities
- mass conservation
- convergence (never turned out to be an issue in these experiments)

Higher values of the parameter turned out to cause mass losses and spurious velocities at the same time. Lowering the parameter value decreased the mass losses steadily. From these aspects, it would make sense, to allow even the smallest subdivisions of elements.
On the other hand, for an infinitively small subdivision, we find p = p_h and p_enr = 0. If p_h and u_h already satisfy res_enr = 0 (such as e.g. in a hydrostatic case), then the matrix K_enr (is equal to ( d res_enr / d p_enr ) ) is approximately a zero-matrix. The inversion of such a matrix causes problems or at least inexactness in the static condensation. For such a case, the algorithm provides a solution, that is applied of the "min_area_ratio" is not reached.

For the given reasons, I would suggest to choose a small value that, however, is not zero.